ogusa/tests/test_aggregates.py

```diff
@@ -73,23 +78,183 @@ def shifted_arr():
+                          'SS'))
     assert (np.allclose(aggI_SS, aggI_SS_test))
 
+    # test TPI
     aggI_TPI_test = ((1 + g_n) * np.exp(g_y) *
                      (K_p1 - res_matrix.sum(1).sum(1) / (1 + g_n)) -
                      (1.0 - delta) * K)
     aggI_TPI = aggr.get_I(b_splus1, K_p1, K,
                           (delta, g_y, omega, lambdas, imm_rates, g_n, 'TPI'))
     assert (np.allclose(aggI_TPI, aggI_TPI_test))
+
+
+def test_get_K():
+    """
+    Simulate data similar to observed
+    """
+    T = 160
+    s, j = 40, 2
+
+    b = -0.1 + (7 * np.random.rand(T * s * j).reshape(T, s, j))
+    omega = 0.5 * np.random.rand(T * s).reshape(T, s, 1)
+    lambdas = 0.4 + (0.2 * np.random.rand(j).reshape(1, 1, j))
+    g_n = 0.1 * np.random.rand(T)
+    imm_rates = -0.1 + np.random.rand(T * s * 1).reshape(T, s, 1)
+
+    omega_extended = np.append(omega[:, 1:, :], np.zeros((T, 1, 1)), axis=1)
+    imm_extended = np.append(imm_rates[:, 1:, :], np.zeros((T, 1, 1)), axis=1)
+
+    K_test = ((b * omega * lambdas) +
+              (b * (omega_extended * imm_extended) * lambdas))
+    K = aggr.get_K(b[0], (omega[0], lambdas[0], imm_rates[0], g_n[0], "SS"))
+
+    assert np.allclose(K_test[0].sum()/(1.0 + g_n[0]), K)
+
+    K = aggr.get_K(b, (omega, lambdas, imm_rates, g_n, "TPI"))
+    assert np.allclose(K_test.sum(1).sum(1)/(1.0 + g_n), K)
+
+
+def test_get_BQ():
+    """
+    Simulate data similar to observed
+    """
+    T = 160
+    s, j = 40, 2
+
+    r = 0.5 + 0.5 * np.random.rand(T).reshape(T, 1)
+    b_splus1 = 0.06 + 7 * np.random.rand(T, s, j)
+    omega = 0.5 * np.random.rand(T, s, 1)
+    lambdas = 0.4 + 0.2 * np.random.rand(j).reshape(1,1,j)
+    rho = np.random.rand(s).reshape(1,s,1)
+    g_n = 0.1 * np.random.rand(T).reshape(T, 1)
+
+    BQ_presum = b_splus1 * omega * rho * lambdas
+    factor = (1.0 + r) / (1.0 + g_n)
+
+    # test SS
+    BQ = aggr.get_BQ(r[0], b_splus1[0],
+                     (omega[0], lambdas[0], rho[0], g_n[0], "SS"))
+    assert np.allclose(BQ_presum[0].sum(0) * factor[0], BQ)
+
+    # test TPI
+    BQ = aggr.get_BQ(r, b_splus1,
+                     (omega, lambdas, rho, g_n, "TPI"))
+    assert np.allclose(BQ_presum.sum(1) * factor, BQ)
+
+
+def test_get_C():
+    """
+    Simulate data similar to observed
+    """
+    T = 160
+    s, j = 40, 2
+
+    c = 0.1 + 0.5 * np.random.rand(T * s * j).reshape(T, s, j)
+    omega = 0.5 * np.random.rand(T * s).reshape(T, s, 1)
+    lambdas = np.random.rand(j)
+
+    aggC_presum = c * omega * lambdas
+
+    # test SS
+    aggC = aggr.get_C(c[0], (omega[0], lambdas, "SS"))
+    assert np.allclose(aggC_presum[0].sum(), aggC)
+    # test TPI
+    aggC = aggr.get_C(c, (omega, lambdas, "TPI"))
+    assert np.allclose(aggC_presum.sum(1).sum(1), aggC)
+
+
+def test_revenue():
+    """
+    Simulate data similar to observed and compare current results with saved
+    results
+    """
+    T = 30
+    s, j = 20, 2
```

Comment:
@hdoupe Might use capital letters for S and J (lower case in the code refers to an item in a range from 0 to S (or J), while S/J give the max number in that dimension.

Reply:
Ok, sounds good to me.